packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -238,21 +254,33 @@ public ResponseEntity<UUID> equationsToModel(
 				if (documentId != null) {
 					modelService.enrichModel(projectId, documentId, model.getId(), permission, true);
 				}
-				return ResponseEntity.ok(model.getId());
+				return ResponseEntity.ok(new EquationsToModelResponse(model.getId(), cleanedEquations));
 			} catch (final IOException e) {
 				log.error("An error occurred while trying to retrieve information necessary for model enrichment.", e);
+				if (!cleanedEquations.isEmpty()) {
+					return ResponseEntity.ok(new EquationsToModelResponse(null, cleanedEquations));
+				}
 				throw new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, messages.get("postgres.service-unavailable"));
 			} catch (ExecutionException e) {
 				log.error("Error while waiting for task response", e);
+				if (!cleanedEquations.isEmpty()) {
+					return ResponseEntity.ok(new EquationsToModelResponse(null, cleanedEquations));
+				}
 				throw new ResponseStatusException(
 					HttpStatus.INTERNAL_SERVER_ERROR,
 					messages.get("task.gollm.execution-failure")
 				);
 			} catch (InterruptedException e) {
 				log.warn("Interrupted while waiting for task response", e);
+				if (!cleanedEquations.isEmpty()) {
+					return ResponseEntity.ok(new EquationsToModelResponse(null, cleanedEquations));
+				}
 				throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, messages.get("task.gollm.interrupted"));
 			} catch (TimeoutException e) {
 				log.warn("Timeout while waiting for task response", e);
+				if (!cleanedEquations.isEmpty()) {
+					return ResponseEntity.ok(new EquationsToModelResponse(null, cleanedEquations));
+				}
```

Comment:
This pattern of returning `cleanedEquations` seems very strange.

If we are always going to return it then we should ignore the other Java exception and just return this, instead of returning it for every single type of exception? 

I don't exactly know what logic we want to have here, but this feels like an anti-pattern.

Reply:
We checked in with gauldie - it is probably better to split this into two steps than try to around both returning results and exception handling.

- cleaned equation
- the skema stuff